### ISSUE TITLE
[Bug] Flag color or Mark in Reviewer isn't updated if changed in CardBrowser

### ISSUE THREAD

**Issue by dorrin-sot** [ankidroid/Anki-Android#11396]
###### Reproduction Steps

1. Start reviewing a card
2. Open Drawer
3. Select Card Browser
4. Change the flag of the card that you were reviewing in step 1 (Issue also exists for Marking/Unmarking the card)
5. Go back to reviewer (By pressing back)

###### Expected Result
The card in the reviewer is not updated

###### Actual Result
The card in the reviewer is updated

###### Debug info
```
AnkiDroid Version = 2.16alpha65

Android Version = 11

Manufacturer = unknown

Model = Android SDK built for x86

Hardware = ranchu

Webview User Agent = Mozilla/5.0 (Linux; Android 11; Android SDK built for x86 Build/RSR1.210210.001.A1; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/83.0.4103.106 Mobile Safari/537.36

ACRA UUID = d8820739-7e9b-4fce-a8c2-4f44d5bda21e

Scheduler = std2

Crash Reports Enabled = false

DatabaseV2 Enabled = true

```

**Comment by dorrin-sot** [comment]
I would like to work on this if it is accepted

**Comment by BrayanDSO** [comment]
This may be related with #11046. I guess reviewer isn't updated if there are changes on CardBrowser. 

Using `startActivityForResult` (or preferentially the non deprecated method) should work, but would probably result with the same problem as #11046

**Comment by dorrin-sot** [comment]
Actually this is way easier to fix. I fixed it by setting `reloadRequired` to true inside the handlers for `flag` and `mark` in card browser. `suspend` and `bury` may be fixed similarly too🤔

**Comment by BrayanDSO** [comment]
I believe it is easy to fix, but can you please check if the following works to see if the same bug won't happen here:

1. Open the reviewer
2. Open **Statistics** from the navigation drawer
3. Open card browser and mark a card
4. Go back to the reviewer
5. Check if the card is marked

**Comment by dorrin-sot** [comment]
It didn't. But after I also added the "reloadRequired" and "currentCard" in the intents of NavigationDrawer it worked (If you're not already working on that issue I can do some cleanup and commit it)

**Comment by BrayanDSO** [comment]
Go for it 👍🏻

Em sáb., 21 de mai. de 2022 08:07, dorrin-sot ***@***.***>
escreveu:

**Comment by dorrin-sot** [comment]
> 

@BrayanDSO Hi I fixed this (still working on #11046 😅) but I couldn't find an Issue for it specifically
